cmd/server-main.go

```diff
@@ -256,10 +256,32 @@ func initSafeMode() (err error) {
 }
 
 func initAllSubsystems(newObject ObjectLayer) (err error) {
-	// List buckets to be re-used for loading configs.
-	buckets, err := newObject.ListBuckets(GlobalContext)
-	if err != nil {
-		return fmt.Errorf("Unable to list buckets: %w", err)
+	var buckets []BucketInfo
+	if globalIsDistXL || globalIsXL {
+		// List buckets to heal, and be re-used for loading configs.
+		buckets, err = newObject.ListBucketsHeal(GlobalContext)
+		if err != nil {
+			return fmt.Errorf("Unable to list buckets to heal: %w", err)
+		}
+		// Attempt a heal if possible and re-use the bucket names
+		// to reload their config.
+		wquorum := &InsufficientWriteQuorum{}
+		rquorum := &InsufficientReadQuorum{}
+		for _, bucket := range buckets {
+			if err = newObject.MakeBucketWithLocation(GlobalContext, bucket.Name, ""); err != nil {
+				if errors.As(err, &wquorum) || errors.As(err, &rquorum) {
+					return fmt.Errorf("Unable to heal bucket: %w", err)
```

Comment:
From automation's point of view, this is probably little extreme. This will make the server die, right ? is it necessary ?

Reply:
It won't @vadmeste it will be retried